base/src/SceneDescriptorXForm.cpp

```diff
@@ -0,0 +1,174 @@
+  std::string mOutputPinId;
+  SceneDescriptorXFormProps mProps;
+  ModelStrategy::ModelStrategyType modelStrategyType;
+  boost::shared_ptr<ModelStrategy> modelStrategy;
+};
+
+SceneDescriptorXForm::SceneDescriptorXForm(SceneDescriptorXFormProps _props)
+    : Module(TRANSFORM, "SceneDescriptorXForm", _props) {
+  mDetail.reset(new Detail(_props));
+}
+
+SceneDescriptorXForm::~SceneDescriptorXForm() {}
+
+bool SceneDescriptorXForm::validateInputPins() {
+  if (getNumberOfInputPins() != 1) {
+    LOG_ERROR << "<" << getId()
+              << ">::validateInputPins size is expected to be 1. Actual<"
+              << getNumberOfInputPins() << ">";
+    return false;
+  }
+
+  framemetadata_sp metadata = getFirstInputMetadata();
+
+  FrameMetadata::FrameType frameType = metadata->getFrameType();
+
+  if (frameType != FrameMetadata::FrameType::ENCODED_IMAGE) {
+    LOG_ERROR << "<" << getId()
+              << ">::validateInputPins input frameType is expected to be "
+                 "Audio. Actual<"
+              << frameType << ">";
+    return false;
+  }
+
+  FrameMetadata::MemType memType = metadata->getMemType();
+  if (memType != FrameMetadata::MemType::HOST) {
+    LOG_ERROR
+        << "<" << getId()
+        << ">::validateInputPins input memType is expected to be HOST. Actual<"
+        << memType << ">";
+    return false;
+  }
+  return true;
+}
+
+bool SceneDescriptorXForm::validateOutputPins() {
+  if (getNumberOfOutputPins() != 1) {
+    LOG_ERROR << "<" << getId()
+              << ">::validateOutputPins size is expected to be 1. Actual<"
+              << getNumberOfOutputPins() << ">";
+    return false;
+  }
+
+  framemetadata_sp metadata = getFirstOutputMetadata();
+  FrameMetadata::FrameType frameType = metadata->getFrameType();
+  if (frameType != FrameMetadata::FrameType::TEXT) {
+    LOG_ERROR << "<" << getId()
+              << ">::validateOutputPins input frameType is expected to be "
+                 "TEXT. Actual<"
+              << frameType << ">";
+    return false;
+  }
+
+  return true;
+}
+
+void SceneDescriptorXForm::addInputPin(framemetadata_sp &metadata,
+                                       string &pinId) {
+  Module::addInputPin(metadata, pinId);
+  mDetail->mOutputMetadata =
+      framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::TEXT));
+  mDetail->mOutputMetadata->copyHint(*metadata.get());
+  mDetail->mOutputPinId = addOutputPin(mDetail->mOutputMetadata);
+}
+
+bool SceneDescriptorXForm::init() {
+  bool ret = mDetail->modelStrategy->initStrategy();
+  if (!ret) {
+    return false;
+  }
+  return Module::init();
+}
+
+bool SceneDescriptorXForm::term() {
+  bool ret = mDetail->modelStrategy->termStrategy();
+  if (!ret) {
+    return false;
+  }
+  return Module::term();
+}
+
+bool SceneDescriptorXForm::process(frame_container &frames) {
+  /*Encoder Model*/
+  mDetail->modelStrategy->encoderModel->push(frames);
+  mDetail->modelStrategy->encoderModel->step();
```

Comment:
Opt 1
-> Push Should Internally call step
Opt 2
-> If Que is not empty , then we should automatically take up the job

Reply:
Done